api/sip/__init__.py

```diff
@@ -84,33 +84,39 @@ def __init__(self, library, integration, analytics=None, client=None, connect=Tr
         super(SIP2AuthenticationProvider, self).__init__(
             library, integration, analytics
         )
+
+        self.server = integration.url
+        self.port = integration.setting(self.PORT).int_value
+        self.login_user_id = integration.username
+        self.login_password = integration.password
+        self.location_code = integration.setting(self.LOCATION_CODE).value
+        self.field_separator = integration.setting(self.FIELD_SEPARATOR).value or '|'
+        self.client = client
+
+    def patron_information(self, username, password):
         try:
-            server = None
-            if client:
-                if callable(client):
-                    client = client()
+            if self.client:
+                sip = self.client
```

Comment:
The `if callable` code should still be around, so that we can test by passing in a mock class rather than an instance, but it can be moved to line 94.

Reply:
Do we actually need this though? All the tests work by passing in an instance?

Reply:
You're right, if the tests are passing we don't need it.